superset/utils/dashboard_filter_scopes_converter.py

```diff
@@ -77,10 +77,13 @@ def copy_filter_scopes(
 ) -> Dict:
     new_filter_scopes = {}
     for (slice_id, scopes) in old_filter_scopes.items():
-        new_filter_key = old_to_new_slc_id_dict[int(slice_id)]
-        new_filter_scopes[str(new_filter_key)] = scopes
-        for scope in scopes.values():
-            scope["immune"] = [
-                old_to_new_slc_id_dict[slice_id] for slice_id in scope.get("immune")
-            ]
+        new_filter_key = old_to_new_slc_id_dict.get(int(slice_id))
+        if new_filter_key:
+            new_filter_scopes[str(new_filter_key)] = scopes
+            for scope in scopes.values():
+                scope["immune"] = [
+                    old_to_new_slc_id_dict[slice_id]
+                    for slice_id in scope.get("immune")
+                    if old_to_new_slc_id_dict.get(slice_id) is not None
```

Comment:
Actually I don't think line 87 is required since this is `new_filter_key` which on line 81 you already tested its truthiness.

Reply:
it is necessary.  line 81 is check filter slice is still in the dashboard, line 87 is to check if immune charts still in current dashboard.

Reply:
Apologies. I didn't comprehend that `slice_id` was a local variable in the list comprehension. It's still not clear if `old_to_new_slc_id_dict[slice_id]` can be `None` and hence whether the logic on line 87 is correct or can be simplified to my previous comment.

Reply:
`old_to_new_slc_id_dict[int(slice_id)]` should not be None as long as slice_id in the `old_to_new_slc_id_dict`